src/contracts/signer-identity.spec.ts

```diff
@@ -0,0 +1,364 @@
+/*
+  Copyright (c) 2018-present evan GmbH.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import 'mocha';
+import BigNumber = require('bignumber.js');
+import chaiAsPromised = require('chai-as-promised');
+import { expect, use } from 'chai';
+import {
+  ContractLoader,
+  Executor,
+  SignerInternal,
+} from '@evan.network/dbcp';
+
+import { accounts } from '../test/accounts';
+import { SignerIdentity } from './signer-identity';
+import { TestUtils } from '../test/test-utils';
+
+
+use(chaiAsPromised);
+
+describe('signer-identity (identity based signer)', function() {
+  this.timeout(300000);
+
+  let executor: Executor;
+  let signer;
+  let web3;
+
+  before(async () => {
+    web3 = TestUtils.getWeb3();
+
+    const contracts = await TestUtils.getContracts();
+    const contractLoader =  new ContractLoader({
+      contracts,
+      web3,
+    });
+    const accountStore = TestUtils.getAccountStore({});
+    const verifications = await TestUtils.getVerifications(web3, await TestUtils.getIpfs());
+    const underlyingSigner = new SignerInternal({
+      accountStore,
+      contractLoader,
+      config: {},
+      web3,
+    });
+    signer = new SignerIdentity(
+      {
+        contractLoader,
+        verifications,
+        web3,
+      },
+      {
+        activeIdentity: await verifications.getIdentityForAccount(accounts[3], true),
+        underlyingAccount: accounts[3],
+        underlyingSigner,
+      },
+    );
+    executor = new Executor(
+      { config: { alwaysAutoGasLimit: 1.1 }, signer: signer, web3 });
+    await executor.init({ eventHub: await TestUtils.getEventHub(web3) });
+
```

Comment:
```suggestion
```

Reply:
Empty lines at the end of files is an older but broadly used POSIX/Unix convention. We have this in most api project files.